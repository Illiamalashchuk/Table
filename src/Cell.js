// @flow
import React from 'react';


type CellProps = {
  isEnterForPercent: boolean,
  highlight: boolean,
  id: number,
  value: string | number,
  changeValues: (event: SyntheticEvent<HTMLTableCellElement>) => void,
  lingthUpSomeValues: (event: SyntheticEvent<HTMLTableCellElement>) => void,
  clearLightedValues: () => void
}

export default function Cell(props: CellProps) {
  const { isEnterForPercent, highlight, id, changeValues, lingthUpSomeValues, clearLightedValues, value } = props;
  const className = isEnterForPercent ? ('table-light') : ((highlight) ? ('table-primary') : ('table-light'));
  return (
    <td
      className={className}
      data-id={id}
      onClick={changeValues}
      onMouseOver={lingthUpSomeValues}
      onMouseOut={clearLightedValues}
    >
      {value}
      <div
        style={isEnterForPercent ? ({ backgroundColor: 'blue', paddingTop: value }) : ({ display: 'none' })}
        />
    </td>
  );
}
