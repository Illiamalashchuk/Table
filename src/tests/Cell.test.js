import React from 'react';
import { shallow } from 'enzyme';
import Cell from '../Cell';
import renderer from 'react-test-renderer';


import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

describe('Cell component', () => {
  it('unhighlighted status', () => {
    const cell = renderer
    .create(<Cell highlight={false} isEnterForPercent={false}/>)
    .toJSON()
    expect(cell).toMatchSnapshot();
  });

  it('highlighted status', () => {
    const cell = renderer
    .create(<Cell highlight={true} isEnterForPercent={false}/>)
    .toJSON()
    expect(cell).toMatchSnapshot();
  });

  it('highlighted for percent', () => {
    const cell = renderer
    .create(<Cell highlight={false} isEnterForPercent={true}/>)
    .toJSON()
    expect(cell).toMatchSnapshot();
  });

});

