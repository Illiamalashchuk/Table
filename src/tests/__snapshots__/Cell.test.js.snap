// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cell component highlighted for percent 1`] = `
<td
  className="table-light"
  data-id={undefined}
  onClick={undefined}
  onMouseOut={undefined}
  onMouseOver={undefined}
>
  <div
    style={
      Object {
        "backgroundColor": "blue",
        "paddingTop": undefined,
      }
    }
  />
</td>
`;

exports[`Cell component highlighted status 1`] = `
<td
  className="table-primary"
  data-id={undefined}
  onClick={undefined}
  onMouseOut={undefined}
  onMouseOver={undefined}
>
  <div
    style={
      Object {
        "display": "none",
      }
    }
  />
</td>
`;

exports[`Cell component unhighlighted status 1`] = `
<td
  className="table-light"
  data-id={undefined}
  onClick={undefined}
  onMouseOut={undefined}
  onMouseOver={undefined}
>
  <div
    style={
      Object {
        "display": "none",
      }
    }
  />
</td>
`;
